File: RockitCompiler/benchmarks/bench_file_read.js
const fs = require('fs');
const content = fs.readFileSync(process.argv[2], 'utf8');
let lines = 0;
let bytes = Buffer.byteLength(content, 'utf8');
for (let i = 0; i < content.length; i++) {
    if (content[i] === '\n') lines++;
}
console.log(lines + " " + bytes);

File: RockitCompiler/benchmarks/bench_file_read.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	f, err := os.Open(os.Args[1])
	if err != nil {
		panic(err)
	}
	defer f.Close()

	scanner := bufio.NewScanner(f)
	scanner.Buffer(make([]byte, 1024*1024), 1024*1024)
	lines := 0
	bytes := 0
	for scanner.Scan() {
		lines++
		bytes += len(scanner.Bytes()) + 1 // +1 for newline
	}
	fmt.Printf("%d %d\n", lines, bytes)
}
